### PR TITLE
Reusability

### PR DESCRIPTION
**Pull request by joshmoore**
Looking to be able to use this repo locally as well as on travis. Current modifications are for testing the 6.0.0-* series on OSX. Current test script:

```
export URL=file:///tmp/update
export LOGIN=ssh:jamoore@localhost
export DIR=/tmp/update
export IJ_PLATFORM=macosx
export IJ_LAUNCHER=$HOME/Fiji.app/Contents/MacOS/ImageJ-macosx
./travis-deploy.sh
```

### CODE REVIEW

File: travis-deploy.sh [hunk 1]
```diff
@@ -1,26 +1,35 @@
...
 # Install ImageJ
-mkdir -p $IJ_PATH/
-cd $HOME/
-wget --no-check-certificate https://downloads.imagej.net/fiji/latest/fiji-linux64.zip
-unzip fiji-linux64.zip
-$IJ_LAUNCHER --update update-force-pristine
+if [ ! -e "$IJ_PATH" ];
+then
+    mkdir -p "$IJ_PATH/"
+    pushd $HOME/
```
Comment (sbesson) [inline travis-deploy.sh:22]: Getting `./travis-deploy.sh: 22: ./travis-deploy.sh: pushd: not found` - see https://travis-ci.org/sbesson/bio-formats-fiji/builds/469604591#L1734

**Comment by sbesson** [comment]
https://github.com/joshmoore/bio-formats-fiji/pull/1 opened with the fix for https://github.com/ome/bio-formats-fiji/pull/19#discussion_r242618604.

This has been successfully deployed to https://sites.imagej.net/Sbesson/jars/bio-formats/ via https://travis-ci.org/sbesson/bio-formats-fiji/builds/470084400#L8743.

Enabling the update site in my local Fiji distribution correctly upgraded the packages but opening an image failed immediately with

```
(Fiji Is Just) ImageJ 2.0.0-rc-69/1.52i; Java 1.8.0_66 [64-bit]; Mac OS X 10.13.6; 206MB of 9406MB (2%)
 
java.lang.NoClassDefFoundError: io/minio/errors/MinioException
	at loci.common.Location.getHandle(Location.java:488)
	at loci.common.Location.getHandle(Location.java:462)
	at loci.common.Location.getHandle(Location.java:443)
	at loci.common.Location.getHandle(Location.java:426)
	at loci.common.Location.checkValidId(Location.java:551)
	at loci.formats.ImageReader.getReader(ImageReader.java:177)
	at loci.plugins.in.ImportProcess.createBaseReader(ImportProcess.java:620)
	at loci.plugins.in.ImportProcess.initializeReader(ImportProcess.java:485)
	at loci.plugins.in.ImportProcess.execute(ImportProcess.java:138)
	at loci.plugins.in.Importer.showDialogs(Importer.java:140)
	at loci.plugins.in.Importer.run(Importer.java:76)
	at loci.plugins.LociImporter.run(LociImporter.java:78)
	at ij.IJ.runUserPlugIn(IJ.java:228)
	at ij.IJ.runPlugIn(IJ.java:192)
	at ij.Executer.runCommand(Executer.java:137)
	at ij.Executer.run(Executer.java:66)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.ClassNotFoundException: io.minio.errors.MinioException
	at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:331)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	... 17 more
```

due to the absence of the `minio` dependencies. A hacky workaround would be to push `minio` to the update site as well but we might want to look into:

- making it more optional i.e. effectively more `runtime`
- opening the discussion to add it to the base sets of JARs for Bio-Formats 6

**Comment by sbesson** [comment]
See https://trello.com/c/UJGyYSrg/129-s3-and-imagej. For rc1, the objective is to fix this blocker at the Bio-Formats I/O level without forcing the addition  of an extra 3d party dependency in the update site.

**Comment by sbesson** [comment]
Last commit should bump to rc1. Proposing to merge for deployment on http://sites.imagej.net/Bio-Formats/ /cc @dgault @joshmoore

**Review by dgault — APPROVED** [review]

**Pull request merged**